ibis/common/bases.py

```diff
@@ -86,7 +121,7 @@ class Singleton(Base):
 
     @classmethod
     def __create__(cls, *args, **kwargs):
-        key = (cls, args, FrozenDict(kwargs))
+        key = (cls, args, tuple(kwargs.items()))
```

Comment:
`ibis.common.collections` is now depending on `ibis.common.bases` so `FrozenDict` would cause a circular import.